examples/fluids/src/setupstg_shur14.c

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2017, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-734707. All Rights
+// reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+/// @file
+/// Implementation of the Synthetic Turbulence Generation (STG) algorithm
+/// presented in Shur et al. 2014
+
+#include <stdlib.h>
+#include <math.h>
+#include "../qfunctions/stg_shur14.h"
+
+#ifndef M_PI
+#define M_PI    3.14159265358979323846
+#endif
+
+#ifndef NOMINMAX
+
+#ifndef max
+#define max(a,b)            (((a) > (b)) ? (a) : (b))
+#endif
+
+#ifndef min
+#define min(a,b)            (((a) < (b)) ? (a) : (b))
+#endif
+
+#endif  /* NOMINMAX */
+
+/*
+ * @brief Perform Cholesky decomposition on array of symmetric 3x3 matrices
+ *
+ * This assumes the input matrices are in order [11,22,33,12,13,23]. This
+ * format is also used for the output.
+ *
+ * @param[in] nprofs Number of matrices in Rij
+ * @param[in] Rij Array of the symmetric matrices [6,nprofs]
+ * @param[out] Cij Array of the Cholesky Decomposition matrices, [6,nprofs]
+ */
+static inline int CalcCholeskyDecomp(int nprofs, const CeedScalar Rij[6][nprofs], CeedScalar Cij[6][nprofs]){
+
+  CeedPragmaSIMD
+  for(int i=0; i<nprofs; i++){
+    Cij[0][i] = sqrt(Rij[0][i]);
+    Cij[3][i] = Rij[3][i] / Cij[0][i];
+    Cij[1][i] = sqrt(Rij[1][i] - pow(Cij[3][i], 2) );
+    Cij[4][i] = Rij[4][i] / Cij[0][i];
+    Cij[5][i] = (Rij[5][i] - Cij[3][i]*Cij[4][i]) / Cij[1][i];
+    Cij[2][i] = sqrt(Rij[2][i] - pow(Cij[4][i], 2) - pow(Cij[5][i], 2));
+  }
+return 1;
+}
+
+
+int SetupSTG_Rand(STGShur14Context stg_ctx){
+
+  //TODO will probably want to have the paths to the STGRand.dat and
+  // STGInflow.dat as inputs for this function.
```

Comment:
That or you can use `PetscOptionsString` and the like so it can be specified via command-line or yaml file.